New version: Microsoft.PIX version 24.3.8.1

Dragon1573 opened the pull request:
## Checklist for Pull Requests

- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] There's a linked pull request. Resolve #146860

## Manifests

- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
  - There is #135219 but failed to merge, as they changed the installer type.
- [x] This PR only modifies one (1) manifest
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/doc/Authoring.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
  - Only `x64` installer file is verified in Windows Sandbox.
- [x] Does your manifest conform to the [1.6 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.6.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/146944)

Dragon1573 committed:
New version: Microsoft.PIX version 24.3.8.1

Signed-off-by: 是蓝兔哟～ <49941141+Dragon1573@users.noreply.github.com>

Trenly commented on manifests/m/Microsoft/PIX/24.3.8.1/Microsoft.PIX.locale.en-US.yaml:
```diff
@@ -0,0 +1,18 @@
+Author: Microsoft
+PackageName: Microsoft PIX 2403.08
+PackageUrl: https://devblogs.microsoft.com/pix/download/
+License: Copyright (c) Microsoft Corporation. All rights reserved.
```
A copyright isn't a license; Please update this with actual license terms such as Proprietary, MIT, GPL-2.0, or other terms as appropriate

Trenly commented on manifests/m/Microsoft/PIX/24.3.8.1/Microsoft.PIX.installer.yaml:
```diff
@@ -0,0 +1,24 @@
+Scope: machine
+ReleaseDate: 2024-03-29
+UpgradeBehavior: uninstallPrevious
+InstallerSwitches:
+  Custom: /norestart
+  Log: /log <LOGPATH>
+  SilentWithProgress: /passive
+  Silent: /quiet
```
These are not necessary, as the installer type is `burn`

Dragon1573 replied:
```suggestion
```

`InstallerSwitches` should be removed for `burn` installer type.

Trenly commented on manifests/m/Microsoft/PIX/24.3.8.1/Microsoft.PIX.installer.yaml:
```diff
@@ -0,0 +1,24 @@
+
+PackageIdentifier: Microsoft.PIX
+PackageVersion: 24.3.8.1
+InstallerType: exe
```
```suggestion
InstallerType: burn
```

Trenly reviewed (changes requested):
This PR also does not satisfy #146860 as it does not require the EULA acceptance

Trenly commented on manifests/m/Microsoft/PIX/24.3.8.1/Microsoft.PIX.locale.en-US.yaml:
```diff
@@ -0,0 +1,18 @@
+PackageLocale: en-US
+Publisher: Microsoft Corporation
+Author: Microsoft
+PackageName: Microsoft PIX 2403.08
```
```suggestion
PackageName: Microsoft PIX
```
Including the version here will cause other versions to not be mapped correctly

sfreed141 commented on manifests/m/Microsoft/PIX/24.3.8.1/Microsoft.PIX.installer.yaml:
```diff
@@ -0,0 +1,24 @@
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.6.0.schema.json
+
+PackageIdentifier: Microsoft.PIX
+PackageVersion: 24.3.8.1
```
Please update the versioning scheme to be consistent with PIX. The correct version number is 2403.08

Trenly replied:
@sfreed141 - To do that all other versions of PIX will need to be removed and re-added to the repository to ensure the correct sorting order.  Using this version scheme is currently necessary as that is what PIX writes to Add and Remove Programs

![image](https://github.com/microsoft/winget-pkgs/assets/12611259/d605ca32-e6e3-4c7d-a284-dc3b36e5d22d)

sfreed141 replied:
Ah okay, thanks for the explanation!

Dragon1573 replied:
> [!NOTE]
>
> Because of the following reason, this change will not be applied.
>
> - See https://github.com/microsoft/winget-pkgs/pull/146944#discussion_r1544686470

sfreed141 commented:
@Dragon1573 thanks for the quick PR! However I think there was a misunderstanding: we _want_ the EULA to be shown and require users to accept it before installation.

Also, the version number should ideally match PIX's versioning scheme. Instead of 24.3.8.1 it should be 2403.08.

Dragon1573 committed:
Fix (installer.yaml): Change installer type from exe to burn

The installer file is packaged with Burn and `InstallerSwitches` is not necessary.

Co-authored-by: Kaleb Luedtke <trenlymc@gmail.com>

Dragon1573 committed:
Fix (locale.en-US.yaml): PackageName

- Remove the exact version from `PackageName`.
- This may cause problems that other versions can't be mapped.

Co-authored-by: Kaleb Luedtke <trenlymc@gmail.com>

Dragon1573 commented on manifests/m/Microsoft/PIX/24.3.8.1/Microsoft.PIX.locale.en-US.yaml:
```diff
@@ -0,0 +1,18 @@
+Author: Microsoft
+PackageName: Microsoft PIX
+PackageUrl: https://devblogs.microsoft.com/pix/download/
+License: Copyright (c) Microsoft Corporation. All rights reserved.
```
```suggestion
License: Proprietary
```

Unable to find the opensource license from the `PackageUrl`. Maybe it's a proprietary package authored by _Microsoft_.

Dragon1573 committed:
Fix (locale.en-US.yaml): License

- Unable to find the opensource license file.
- Configure this package as a proprietary package from Microsoft.

Trenly commented:
[Policy] Reset Feedback

Trenly commented:
> Also, the version number should ideally match PIX's versioning scheme. Instead of 24

@stephengillie - This package should be a `should` in the auth from @sfreed141 as they are a member of the PIX team. This comment from Sam remains un-addressed

> @Dragon1573 thanks for the quick PR! However I think there was a misunderstanding: we _want_ the EULA to be shown and require users to accept it before installation.
> 
> Also, the version number should ideally match PIX's versioning scheme. Instead of 24.3.8.1 it should be 2403.08.

<!--
[Policy] Needs Attention
-->

Devaniti commented:
I wanted to submit same update and can provide some details
1. You can not change version numbers to match official site. That will fail automatic checks, as PIX, when installed, specifies version as 24.3.8.1 in the Installed apps list.
2. To the best of my knowledge, there is no EULA available on the internet. The only way to see EULA for PIX is to manually open the installer. So there's no URL to specify.
